pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -1109,6 +1115,105 @@ func TestDesiredRouterDeploymentClientTLS(t *testing.T) {
 	}
 }
 
+// TestDesiredRouterDeploymentDynamicConfigManager verifies that desiredRouterDeployment
+// returns the expected deployment when DCM featuregate is involved.
+func TestDesiredRouterDeploymentDynamicConfigManager(t *testing.T) {
+	testCases := []struct {
+		name                       string
+		unsupportedConfigOverrides string
+		dcmEnabled                 bool
+		expectedEnv                []envData
+	}{
+		{
+			name:       "featuregate-enabled-default-values",
+			dcmEnabled: true,
+			expectedEnv: []envData{
+				{"ROUTER_HAPROXY_CONFIG_MANAGER", true, "true"},
+				{"ROUTER_MAX_DYNAMIC_SERVERS", true, "1"},
+				{"ROUTER_BLUEPRINT_ROUTE_POOL_SIZE", true, "0"},
+			},
+		},
+		{
+			name:                       "featuregate-enabled-max-servers",
+			unsupportedConfigOverrides: `{"maxDynamicServers":"7"}`,
```

Comment:
See https://github.com/openshift/cluster-ingress-operator/pull/1159/files#r1844266657

Do we want to allow more than 1 server without documenting the consequences?

Reply:
Right, we should document the consequences. It may be a little difficult for an unsupported option though. Let me think where I can put some words about the consequences.

Reply:
Unsupported config overrides should *not* be documented, at least not in product documentation for end-users.